file/writable_file_writer.h

```diff
@@ -160,7 +160,9 @@ class WritableFileWriter {
   bool perform_data_verification_;
   uint32_t buffered_data_crc32c_checksum_;
   bool buffered_data_with_checksum_;
+#ifndef ROCKSDB_LITE
```

Comment:
Simply adding unused or maybe-used attributes will trigger -Werror=attributes, like below:
```
./file/writable_file_writer.h:163:31: error: ‘maybe_unused’ attribute ignored [-Werror=attributes]
  163 |   [[maybe_unused]]Temperature temperature_;
```

Not sure whether it's okay to remove -Werror=attributes in LITE build so I decide to go for a more verbose way (conditionally declaration and definition)

Reply:
another option is using `(void)temperature_` in LITE.